modules/cassandra/src/main/java/org/testcontainers/cassandra/delegate/CassandraDatabaseDelegate.java

```diff
@@ -0,0 +1,83 @@
+package org.testcontainers.cassandra.delegate;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.testcontainers.cassandra.CassandraContainer;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.ContainerState;
+import org.testcontainers.containers.ExecConfig;
+import org.testcontainers.delegate.AbstractDatabaseDelegate;
+import org.testcontainers.ext.ScriptUtils.ScriptStatementFailedException;
+
+import java.io.IOException;
+
+/**
+ * Cassandra database delegate
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class CassandraDatabaseDelegate extends AbstractDatabaseDelegate<Void> {
+
+    private final ContainerState container;
+
+    @Override
+    protected Void createNewConnection() {
+        // Return null here, because we run scripts using cqlsh command directly in the container.
+        // So, we don't use connection object in the execute() method.
+        return null;
+    }
+
+    @Override
+    public void execute(
+        String statement,
+        String scriptPath,
+        int lineNumber,
+        boolean continueOnError,
+        boolean ignoreFailedDrops
+    ) {
+        try {
+            // Use cqlsh command directly inside the container to execute statements
+            // See documentation here: https://cassandra.apache.org/doc/stable/cassandra/tools/cqlsh.html
+            String[] cqlshCommand = new String[] { "cqlsh" };
+
+            if (this.container instanceof CassandraContainer) {
+                CassandraContainer cassandraContainer = (CassandraContainer) this.container;
+                String username = cassandraContainer.getUsername();
+                String password = cassandraContainer.getPassword();
+                cqlshCommand = ArrayUtils.addAll(cqlshCommand, "-u", username, "-p", password);
```

Comment:
are username and password required to execute scripts, even if the client is not using them? I haven't seen usage of `getUsername` and `getPassword` in tests

Reply:
@eddumelendez 
By default, Cassandra is configured to allow the connections without authentication using the authenticator `AllowAllAuthenticator`. But, if the default configuration of the image is overridden to use the `PasswordAuthenticator`, the username/password becomes mandatory. So, it's better to keep this to take such a case into account. Consequently, I added a test case for that.
There is a little restriction to this: sometimes, it can take few seconds after the database port is listening to allow authenticated queries. That's why I added recommendations for using `CassandraQueryWaitStrategy` (only when authentication is required) in the Javadoc of `withInitScript` method.

Reply:
thanks for the explanation, @maximevw ! I just noticed it happens the same in existing implementation but lacks of documentation. I think would be nice to make the wait strategy smart enough. So, consumers don't worry about it. Can we add a note about it in the docs, please?

Reply:
Ok, I'll move it to the docs as you suggested in your other comment.